google/cloud/spanner/client.cc

```diff
@@ -128,6 +128,37 @@ std::shared_ptr<Connection> MakeConnection(
                                                     std::move(stub));
 }
 
+StatusOr<CommitResult> RunTransaction(
+    Client client, Transaction::ReadWriteOptions const& opts,
+    std::function<TransactionAction(Client, Transaction)> const& f) {
+  Transaction txn = MakeReadWriteTransaction(opts);
+  TransactionAction action;
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  try {
+#endif
+    action = f(client, txn);
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  } catch (...) {
+    client.Rollback(txn);  // ignore error
+    throw;
+  }
+#endif
+  switch (action.action) {
+    case TransactionAction::kCommit: {
+      // TODO(#357): Automatically retry if Commit() aborts. Note: it is
```

Comment:
FYI: we should let a `RetryPolicy` control how many times and/or for how we retry. We already have such policies in the AdminClient.

Reply:
Acknowledged.